### PR TITLE
CL: Create Page for RequestTypeCreate plus tests and stories

### LINKED ISSUE #78: Create RequestTypeCreatePage

### PR DESCRIPTION
**Pull request by chriswlai**
Storybook: https://6734fdad790fe98dc6bf77f1-xbzqlvttjv.chromatic.com/?path=/story/components-requesttype-requesttypeform--create

Dokku deployment: https://proj-rec.dokku-06.cs.ucsb.edu/

NOTE TO STAFF: APPROVE after PR #77 This branch was created off of cl-reqtypes-placeholders because the work there was needed to do this work. After that PR is approved, rebase to branch from main instead.

This PR finishes the create page for the RequestType component. It also adds the corresponding tests and story files. These tasks continue the process of adding RequestType to the frontend. Ultimately, this works towards having RequestType fully integrated into our application

To Test:
The page should look like this at first:
Screenshot 2024-11-23 at 5 21 58 PM

Don't input anything, just press create, it should look like this:
Screenshot 2024-11-23 at 5 22 03 PM

After you fill out all fields and press create, there should be a pop up on the top right similar to the one below:
Screenshot 2024-11-23 at 5 22 11 PM

Related to Issue #42 and Closes #78

### CODE REVIEW

**Review by vprchlik — DISMISSED** [review]
LGTM

**Review by skanda-vasishta — APPROVED** [review]
LGTM!

File: frontend/src/main/components/RequestType/RequestTypeForm.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ function RequestTypeForm({
         </Form.Group>
       )}
 
-      <Form.Group className="mb-3">
+      {/* <Form.Group className="mb-3">
```
Comment (pconrad) [inline frontend/src/main/components/RequestType/RequestTypeForm.js:38]: What is "commented out code" doing in a PR that's marked "Ready For Staff"?

And why did both the original PR author and the person that did the code review think this was ok?

It's literally the first thing in the list of PR Criteria.

https://ucsb-cs156.github.io/topics/pull_requests/#pr-checklist

It's kind of astonishing to still see that at this stage.

**Review by pconrad — CHANGES_REQUESTED** [review]
Commented out code, and a poor user experience for the error that comes up when duplicate RequestTypes are entered.

Here's the fix for the error message: 

https://github.com/ucsb-cs156-f24/proj-courses-f24-02/blob/d7ee97a74cc8f99ced08b81f6be1742b9cf9a09d/frontend/src/main/pages/PersonalSchedules/PersonalSchedulesCreatePage.js#L20

The code looks like this (though you'll need to adapt it to your situation; you can't just copy paste this):

```jsx
  const onError = (error) => {
    toast(`Error: ${error.response.data.message}`);
  };

  const mutation = useBackendMutation(
    objectToAxiosParams,
    { onSuccess, onError },
    // Stryker disable next-line all : hard to set up test for caching
    ["/api/personalschedules/all"],
  );
```

The backend is already producing a reasonable error message:

```
  "message": "Request Type with this description already exists"
```

We just need to show that to the end user instead of the generic one.

**Comment by pconrad** [comment]
Closing per slack channel conversation: https://github.com/ucsb-cs156-f24/proj-rec-f24-06/pull/81

**Pull request closed without merging**